### DISCUSSION TITLE
Single line for simple queries?

### DISCUSSION THREAD

**Discussion by mhanberg** [sql-formatter-org/sql-formatter#648] [Q&A]
Is it possible to format simple, short queries onto a single line?

Basically instead of 

```
$ echo "select * from foo" | sql-formatter -l sqlite
select
  *
from
  foo
```

i'd like it to output

```
$ echo "select * from foo" | sql-formatter -l sqlite
select * from foo
```

**Comment by nene** [comment]
There is an old feature request for this: https://github.com/sql-formatter-org/sql-formatter/issues/360

But it's unlikely to happen as I've pretty much stopped any new feature development on sql-formatter.

Instead I've incorporated many of the ideas and addressed lots of the shortcomings of this in [another SQL pretty-printing tool that I've been developing.](https://github.com/nene/prettier-plugin-sql-cst) This one pretty much does what you're asking for: if the input SQL is on a single line and it's short enough, the formatter will keep it on a single line.

**Reply by mhanberg** [reply]
Gotcha, I'll check out the new project
